domain/mocks/route_mock.go

```diff
@@ -0,0 +1,74 @@
+package mocks
+
+import (
+	"context"
+
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/sqs/domain"
+)
+
+type RouteMock struct {
+	CalculateTokenOutByTokenInFunc      func(ctx context.Context, tokenIn types.Coin) (types.Coin, error)
+	ContainsGeneralizedCosmWasmPoolFunc func() bool
+	GetPoolsFunc                        func() []domain.RoutablePool
+	GetTokenOutDenomFunc                func() string
+	PrepareResultPoolsFunc              func(ctx context.Context, tokenIn types.Coin) ([]domain.RoutablePool, math.LegacyDec, math.LegacyDec, error)
+	StringFunc                          func() string
+}
```

Comment:
Note: I ended up not using the mock but I think it is still useful for the future